File: app/src/main/java/com/example/pizzamaniaapp/app/ui/auth/LoginActivity.java
package com.example.pizzamaniaapp.app.ui.auth;

import android.content.Intent;
import android.os.Bundle;
import android.widget.*;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import com.google.firebase.auth.FirebaseAuth;
import com.example.pizzamaniaapp.R;
import com.example.pizzamaniaapp.app.ui.main.MainActivity;

public class LoginActivity extends AppCompatActivity {
    private FirebaseAuth auth;

    @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        auth = FirebaseAuth.getInstance();
        if (auth.getCurrentUser() != null) {
            startActivity(new Intent(this, MainActivity.class));
            finish();
            return;
        }
        EditText email = findViewById(R.id.etEmail);
        EditText pass = findViewById(R.id.etPassword);
        Button btn = findViewById(R.id.btnLogin);
        TextView goReg = findViewById(R.id.tvGoRegister);
        btn.setOnClickListener(v -> {
            String e = email.getText().toString().trim();
            String p = pass.getText().toString();
            if (e.isEmpty() || p.isEmpty()) { Toast.makeText(this, "Enter email and password", Toast.LENGTH_SHORT).show(); return; }
            btn.setEnabled(false);
            auth.signInWithEmailAndPassword(e, p).addOnCompleteListener(task -> {
                btn.setEnabled(true);
                if (task.isSuccessful()) {
                    startActivity(new Intent(this, MainActivity.class));
                    finish();
                } else {
                    Toast.makeText(this, "Login failed: " + (task.getException()!=null?task.getException().getMessage():""), Toast.LENGTH_LONG).show();
                }
            });
        });
        goReg.setOnClickListener(v -> startActivity(new Intent(this, RegisterActivity.class)));
    }
}

File: app/src/main/java/com/example/pizzamaniaapp/app/ui/auth/LoginActivity.kt
package com.example.pizzamaniaapp.ui

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.button.MaterialButton
import com.google.android.material.textfield.TextInputEditText
import com.google.firebase.auth.ktx.auth
import com.google.firebase.firestore.ktx.firestore
import com.google.firebase.ktx.Firebase
import com.example.pizzamaniaapp.R
import com.example.pizzamaniaapp.ui.admin.AdminDashboardActivity
import com.example.pizzamaniaapp.app.ui.home.HomeActivity
import com.example.pizzamaniaapp.app.ui.auth.RegisterActivity
import com.example.pizzamaniaapp.ui.admin.AdminRegisterActivity

class LoginActivity : AppCompatActivity() {

    private val auth get() = Firebase.auth
    private val db get() = Firebase.firestore

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        // Admin button -> go to AdminRegisterActivity
        findViewById<MaterialButton>(R.id.btnAdmin).setOnClickListener {
            startActivity(Intent(this, AdminRegisterActivity::class.java))
        }

        // User button -> go to User RegisterActivity
        findViewById<MaterialButton>(R.id.btnUser).setOnClickListener {
            startActivity(Intent(this, RegisterActivity::class.java))
        }

        // SIGN IN button
        findViewById<Button>(R.id.btnPrimary).setOnClickListener { view ->
            val email = findViewById<TextInputEditText>(R.id.etEmail).text?.toString()?.trim().orEmpty()
            val pass = findViewById<TextInputEditText>(R.id.etPassword).text?.toString().orEmpty()

            if (email.isEmpty()) { toast("Enter email"); return@setOnClickListener }
            if (pass.isEmpty()) { toast("Enter password"); return@setOnClickListener }

            view.isEnabled = false

            auth.signInWithEmailAndPassword(email, pass)
                .addOnSuccessListener { res ->
                    val uid = res.user?.uid
                    if (uid == null) {
                        view.isEnabled = true
                        toast("Login failed: no user")
                        return@addOnSuccessListener
                    }

                    db.collection("users").document(uid).get()
                        .addOnSuccessListener { doc ->
                            val role = doc.getString("role") ?: "user"
                            if (role == "admin") {
                                startActivity(Intent(this, AdminDashboardActivity::class.java))
                            } else {
                                startActivity(Intent(this, HomeActivity::class.java))
                            }
                            finish()
                        }
                        .addOnFailureListener { e ->
                            view.isEnabled = true
                            toast("Failed to get profile: ${e.message}")
                        }
                }
                .addOnFailureListener { e ->
                    view.isEnabled = true
                    toast(e.message ?: "Login failed")
                }
        }

        // Register text now optional
        findViewById<TextView>(R.id.tvRegister).setOnClickListener {
            toast("Use User button to register 🙂")
        }
    }

    private fun toast(msg: String) =
        Toast.makeText(this, msg, Toast.LENGTH_SHORT).show()
}